File: server/src/main/java/com/github/bondarevv23/memorizer/server/model/Card.java
package com.github.bondarevv23.memorizer.server.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "card")
public class Card {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "deck_id", referencedColumnName = "id")
    private Deck deck;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "question")
    private Side question;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "answer")
    private Side answer;
}

File: bot/src/main/kotlin/memorizer/model/redis/Card.kt
package memorizer.model.redis

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import java.io.Serializable

@RedisHash("Card")
data class Card (
    @Id
    val userId: Long,
    val deckId: Int,
    val question: Side,
    val answer: Side,
    val id: Int? = null
) : Serializable {
    public constructor(userId: Long, deckId: Int) : this(userId, deckId, Side(), Side())
}
